### DISCUSSION TITLE
Get content of message by id

### DISCUSSION THREAD

**Discussion by dhduux** [gotd/td#754] [Q&A]
Hello, can i get the message content by id? Help

**Comment by ghost** [comment]
Try [messages.getMessages](https://core.telegram.org/method/messages.getMessages) for legacy and private chats and [channels.getMessages](https://core.telegram.org/method/channels.getMessages) for channels.

**Comment by dhduux** [comment]
Sorry, can i use it in gotd? Or does it have to be written by hand?

**Reply by ghost** [reply]
[You can call any method defined by Telegram API schema](https://github.com/gotd/td#calling-mtproto-directly)
```go
raw := client.API()
state, err := client.API().MessagesGetMessages(ctx, []tg.InputMessageClass{&tg.InputMessageID{ID: 1}})
if err != nil {
	return err
}
```

[Here example how gotd bot getting a message object by ID.](https://github.com/gotd/bot/blob/b0af8aab12a60fefc909a836451242c4c86956dd/internal/dispatch/base_event.go#L45-L83)

**Reply by ghost** [reply]
> [You can call any method defined by Telegram API schema](https://github.com/gotd/td#calling-mtproto-directly)
> ```go
> raw := client.API()
> state, err := client.API().MessagesGetMessages(ctx, []tg.InputMessageClass{&tg.InputMessageID{ID: 1}})
> if err != nil {
> 	return err
> }
> ```
> 
> [Here example how gotd bot getting a message object by ID.](https://github.com/gotd/bot/blob/b0af8aab12a60fefc909a836451242c4c86956dd/internal/dispatch/base_event.go#L45-L83)
> 

Thank you!!!!
